include/lsst/afw/image/Calib.h

```diff
@@ -69,6 +70,55 @@ inline double fluxErrFromABMagErr(double magErr, double mag) {
     return std::abs(-0.4 * magErr * fluxFromABMag(mag) * std::log(10.0));
 }
 
+
+/// Compute AB magnitude from flux in Janskys
+template <typename T>
+ndarray::Array<T, 1> abMagFromFlux(ndarray::Array<T, 1> const& flux) {
+    ndarray::Array<T, 1> out = ndarray::allocate(flux.getShape());
+    for (std::size_t ii = 0; ii < flux.getNumElements(); ++ii) {
+        out[ii] = abMagFromFlux(flux[ii]);
+    }
+    return out;
+}
+
+/// Compute AB magnitude error from flux and flux error in Janskys
+template <typename T>
+ndarray::Array<T, 1> abMagErrFromFluxErr(ndarray::Array<T, 1> const& fluxErr,
+                                         ndarray::Array<T, 1> const& flux) {
+    if (flux.getNumElements() != fluxErr.getNumElements()) {
+        throw LSST_EXCEPT(pex::exceptions::LengthError, "Length mismatch");
```

Comment:
A more explicit "Length of flux and fluxErr arrays are mismatched: XX vs. YY" might be nice.

Reply:
This was in a header file, and I was both being lazy and wanting to avoid the extra includes that stringifying the lengths requires. I've now moved the definitions out of the header and expanded the error string to include the lengths.